sqlglot/parser.py

```diff
@@ -74,6 +74,17 @@ def _builder(args: t.List, dialect: Dialect) -> E:
     return _builder
 
 
+def build_mod(args: t.List) -> exp.Mod:
+    this = seq_get(args, 0)
+    expression = seq_get(args, 1)
+
+    # Wrap the LHS if it's not a single node, e.g. MOD(a + 1, 7) -> (a + 1) % 7
```

Comment:
We need to do this for both operands, e.g.

```
mysql> SELECT MOD(7, 8 - 1 + 7);
+-------------------+
| MOD(7, 8 - 1 + 7) |
+-------------------+
|                 7 |
+-------------------+
1 row in set (0.00 sec)

mysql> SELECT 7 % 8 - 1 + 7;
+---------------+
| 7 % 8 - 1 + 7 |
+---------------+
|            13 |
+---------------+
1 row in set (0.01 sec)
```

Reply:
True, no idea why I was fixated on LHS